homeassistant/components/weather/__init__.py

```diff
@@ -115,36 +190,84 @@ class WeatherEntity(Entity):
     _attr_ozone: float | None = None
     _attr_precision: float
     _attr_pressure: float | None = None
-    _attr_pressure_unit: str | None = None
+    _attr_pressure_unit: None = None  # Subclasses of WeatherEntity should not set this
+    _attr_native_pressure_unit: str | None = None
     _attr_state: None = None
-    _attr_temperature_unit: str
+    _attr_temperature_unit: None = (
+        None  # Subclasses of WeatherEntity should not set this
+    )
+    _attr_native_temperature_unit: str
     _attr_temperature: float | None
     _attr_visibility: float | None = None
-    _attr_visibility_unit: str | None = None
-    _attr_precipitation_unit: str | None = None
+    _attr_visibility_unit: None = (
+        None  # Subclasses of WeatherEntity should not set this
+    )
+    _attr_native_visibility_unit: str | None = None
+    _attr_precipitation_unit: None = (
+        None  # Subclasses of WeatherEntity should not set this
+    )
+    _attr_native_precipitation_unit: str | None = None
     _attr_wind_bearing: float | str | None = None
     _attr_wind_speed: float | None = None
-    _attr_wind_speed_unit: str | None = None
+    _attr_wind_speed_unit: None = (
+        None  # Subclasses of WeatherEntity should not set this
+    )
+    _attr_native_wind_speed_unit: str | None = None
+
+    _weather_option_temperature_uom: str | None = None
+    _weather_option_pressure_uom: str | None = None
+    _weather_option_visibility_uom: str | None = None
+    _weather_option_precipitation_uom: str | None = None
+    _weather_option_wind_speed_uom: str | None = None
+
+    async def async_internal_added_to_hass(self) -> None:
+        """Call when the sensor entity is added to hass."""
+        await super().async_internal_added_to_hass()
+        if not self.registry_entry:
+            return
+        self.async_registry_entry_updated()
 
     @property
     def temperature(self) -> float | None:
```

Comment:
In a follow-up, after core integrations are migrated, this should be marked with `@final`:
```py
    @final
    @property
    def temperature(self) -> float | None:
```

Reply:
This comment is on the wrong method, it should have been on the `temperature_unit` method.